### PR TITLE
add conda publish github action

### PR DESCRIPTION
**Pull request by felix5572**
add conda publish github action.
it will automatically release dpgen to conda deepmodeling channel  (https://anaconda.org/deepmodeling/repo) when a new tag is added

### CODE REVIEW

File: conda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{% set name = "dpgen" %}
+{% set version = environ.get('GIT_DESCRIBE_TAG').lstrip('v') %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/deepmodeling/dpgen
+  # git_rev: {{ version }}
```
Comment (njzjz) [inline conda/meta.yaml:10]: Why do you comment this line?

File: .github/workflows/publish_conda.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: publish-to-conda
+      uses: felix5572/conda-publish-action@v1.9
+      with:
+        subdir: 'conda'
+        anacondatoken: ${{ secrets.ANACONDA_TOKEN }}
```
Comment (njzjz) [inline .github/workflows/publish_conda.yml:17]: Have you added the token?

File: conda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  number: 0
+  noarch: python
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
+    - git
+  host:
+    - git
```
Comment (njzjz) [inline conda/meta.yaml:21]:
```suggestion
```

Comment (felix5572) [reply conda/meta.yaml]: fixed

File: .github/workflows/publish_conda.yml (hunk 2 above)
Comment (felix5572) [reply .github/workflows/publish_conda.yml:17]: Haven't now, I don't have the permission to add repo secret.  We could add the secret later?

File: conda/meta.yaml (hunk 1 above)
Comment (felix5572) [reply conda/meta.yaml]: typo.. uncomment now

File: conda/construct.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  - defaults
+  - conda-forge
+  - deepmodeling
+
+specs:
+  - python 3.8
+  - pip
+  - dpgen {{ version }} 
+  - dpdata
+  - pymatgen
```
Comment (njzjz) [inline conda/construct.yaml:16]: only dpgen is enough

**Review by AnguseZhang — APPROVED** [review]
I have tested the patch on an independent machine, and things work well.  I have no other comments on the workflow besides Jinzhe's suggestions.

File: conda/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+{% set name = "dpgen" %}
+{% set version = os.popen('git describe --tags --abbrev=0').read().strip('\n').lstrip('v').replace('-', '_') %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/deepmodeling/dpgen
+  git_rev: v{{ version }}
```
Comment (njzjz) [inline conda/meta.yaml:10]: @felix5572 You can consider to use
```suggestion
  path: ..
```
instead. It will fetch source codes from the local file

**Review by njzjz — APPROVED** [review]

**Review by njzjz — APPROVED** [review]
Approve, but note that the secret should be added.

**Pull request merged**